tests/e2e/e2e_statefulsets_test.go

```diff
@@ -0,0 +1,153 @@
+
+				install := action.NewInstall(helmCfg)
+
+				install.ReleaseName = "kafka"
+				install.Namespace = testNS
+				saName := "zookeeper"
+				replicaCount := 3
+
+				cli := cli.New()
+				chartPath, err := install.LocateChart("https://charts.bitnami.com/bitnami/zookeeper-9.0.2.tgz", cli)
+				Expect(err).NotTo(HaveOccurred())
+				chart, err := loader.Load(chartPath)
+				Expect(err).NotTo(HaveOccurred())
+
+				// Install zookeeper
+				_, err = install.Run(chart, map[string]interface{}{
+					"replicaCount": replicaCount,
+					"serviceAccount": map[string]interface{}{
+						"create": true,
+						"name":   saName,
+					},
+				})
+
+				Expect(err).NotTo(HaveOccurred())
+
+				// Create SMI resources for Zookeeper
+				_, err = Td.CreateTCPRoute(testNS, v1alpha4.TCPRoute{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: testNS,
+						Name:      "zookeeper",
+					},
+					Spec: v1alpha4.TCPRouteSpec{
+						Matches: v1alpha4.TCPMatch{
+							Ports: []int{2181, 3181, 2888, 3888},
+						},
+					},
+				})
+				Expect(err).NotTo(HaveOccurred())
+
+				_, err = Td.CreateTrafficTarget(testNS, v1alpha3.TrafficTarget{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: testNS,
+						Name:      "zookeeper",
+					},
+					Spec: v1alpha3.TrafficTargetSpec{
+						Sources: []v1alpha3.IdentityBindingSubject{
+							{
+								Kind:      "ServiceAccount",
+								Name:      saName,
+								Namespace: testNS,
+							},
+						},
+						Destination: v1alpha3.IdentityBindingSubject{
+							Kind:      "ServiceAccount",
+							Name:      saName,
+							Namespace: testNS,
+						},
+						Rules: []v1alpha3.TrafficTargetRule{
+							{
+								Kind: "TCPRoute",
+								Name: "zookeeper",
+							},
+						},
+					},
+				})
+				Expect(err).NotTo(HaveOccurred())
+
+				Expect(Td.WaitForPodsRunningReady(testNS, 90*time.Second, replicaCount, nil)).To(Succeed())
+
+				pods, err := Td.Client.CoreV1().Pods(testNS).List(context.TODO(), metav1.ListOptions{})
+
+				Expect(err).NotTo(HaveOccurred())
+
+				// this command will exit 1 if connectivity isn't established
+				cmd := "/opt/bitnami/zookeeper/bin/zkServer.sh status"
+
+				cond := Td.WaitForRepeatedSuccess(func() bool {
+
+					results := map[string]error{}
+					for _, po := range pods.Items {
+						stdout, stderr, err := Td.RunRemote(testNS, po.GetName(), "zookeeper", strings.Fields(cmd))
+
+						if err != nil {
+							// Error codes from the execution come through err
+							// Curl 'Connection refused' err code = 7
+							Td.T.Logf("> (%s) Remote exec err: %v | stderr: %s", po.GetName(), err, stderr)
+						}
+
+						Td.T.Logf("> (%s) Stdout %s", po.GetName(), stdout)
+
+						results[po.GetName()] = err
+					}
+
+					hadErr := false
+					for podName, err := range results {
+						if err != nil {
+							Td.T.Logf("> (%s) ZK status check failed: expected nil err, got %s", podName, err)
```

Comment:
FWIW it looks like these errors are being logged twice.

Reply:
These are being logged per pod (there are 3 replicas). Are you seeing further duplication?

Reply:
This is the first place I see them logged:
https://github.com/openservicemesh/osm/blob/f3b661f9dc368912bac1e741b4312c3aec0d58c6/tests/e2e/e2e_statefulsets_test.go#L125-L127

Reply:
Ah I see what you're saying. Code has been modified to only log err once 👍🏾